jerry-core/ecma/builtin-objects/ecma-builtin-array-prototype.cpp

```diff
@@ -1477,15 +1477,25 @@ ecma_builtin_array_prototype_object_sort (ecma_value_t this_arg, /**< this argum
     ECMA_FINALIZE (sort_value);
   }
 
-  if (ecma_is_completion_value_empty (ret_value))
+  /* Put values back in the array. */
+  for (uint32_t index = 0; index < copied_num && ecma_is_completion_value_empty (ret_value); index++)
   {
-    /*
-     * FIXME: Casting len to ecma_length_t may overflow, but since ecma_length_t is still at least
-     * 16 bits long, with an array of that size, we would run out of memory way before this happens.
-     */
-    JERRY_ASSERT ((ecma_length_t) len == len);
-    /* Copy the sorted array into a new array. */
-    ret_value = ecma_op_create_array_object (values_buffer, (ecma_length_t) len, false);
+    ecma_string_t *index_string_p = ecma_new_ecma_string_from_uint32 (index);
+
+    if (ecma_is_value_undefined (values_buffer[index]))
+    {
+      ECMA_TRY_CATCH (del_value, ecma_op_object_delete (obj_p, index_string_p, true), ret_value);
```

Comment:
Also, this is current common style. Maybe, this could be encoded in a more convenient way. I think, in the case it is better to change all the places at once.